### ISSUE TITLE
Add date, integer as a field type

### ISSUE THREAD

**Issue by jernsthausen** [datamade/dedupe-api#117]
Journalists often encounter data sets where one can obtain something like birth date or age; could also be helpful where one thinks other date components are important.

**Comment by evz** [comment]
@fgregg Do we have a date comparator? I suppose it would be easy enough to make one.

**Comment by fgregg** [comment]
We don't. I'm not sure what the right behavior is. Maybe we create a date
parserator, and the compare year, month, and day separately?

**Comment by evz** [comment]
Could we just convert it to a Unix timestamp and compare the distance that way?

**Comment by jernsthausen** [comment]
@fgregg That'd be my thinking around dates. But I think initially, we could rely on the user to have dates in a date format and worry about parsing them later.

**Comment by evz** [comment]
On second thought, my approach assumes you actually have relatively consistent / nicely formatted dates to begin with. If we did a parser (using parserator) we could do quite a lot more with messier data.

**Comment by jernsthausen** [comment]
That could work, though I think we'd then want to parse those back into constituent parts ultimately to handle people screwing up on forms. But maybe that wouldn't matter. Might be a question that needs research to answer what's best (and, given how much has been done around the Census and health care, there's probably research out there on this).

**Comment by jernsthausen** [comment]
Also, I do think we are usually going to have nicely formatted dates. User error on the input side is the greater issue, from my perspective.

**Comment by fgregg** [comment]
I don't think we can assume any kind of niceness. We could leverage some
work already done in this area: http://duckling-lib.org/

**Comment by jernsthausen** [comment]
Agreed for longer-term view of making accessible to the average user.

**Comment by fgregg** [comment]
@jernsthausen if you wanted to take a look for some research, that'd be great.

**Comment by jernsthausen** [comment]
@fgregg will do. Just reply with your UChicago library credentials and I'll dive right in.

**Comment by jernsthausen** [comment]
@fgregg just kidding. Obviously. I'll look into it.
